### PR TITLE
[WIP]rimage: test openssl 3.0 integration

### PR DESCRIPTION
**Pull request by juimonen**
Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>

### CODE REVIEW

**Comment by lgirdwood** [comment]
@juimonen looks good to merge into rimage.

**Comment by juimonen** [comment]
@lgirdwood yes, this code flow with < openssl3 should be the same as before refactor... the interesting part is when we get openssl3 in our docker/CI. I tested it should work with couple of platforms, but not sure could we somehow test openssl3 with CI before it is integrated... 

@marc-hb any ideas? So can we test run openssl3 with CI somehow (I could then maybe fix possible issues before integration).

**Comment by lgirdwood** [comment]
> @lgirdwood yes, this code flow with < openssl3 should be the same as before refactor... the interesting part is when we get openssl3 in our docker/CI. I tested it should work with couple of platforms, but not sure could we somehow test openssl3 with CI before it is integrated...
> 
> @marc-hb any ideas? So can we test run openssl3 with CI somehow (I could then maybe fix possible issues before integration).

In this case local testing will do until openSSL is rolled out via the distros. One thing we could do is print the version of openssl in the rimage output so that the version is obvious for any breakage.

We can merge this PR in rimage and have a subsequent PR that prints the openssl version.

**Comment by lgirdwood** [comment]
@juimonen merged into rimage - can you send a submodule PR to update the rimage submodule in SOF.

**Comment by lgirdwood** [comment]
closing as merged into submodule.

**Comment by marc-hb** [comment]
> @marc-hb any ideas? So can we test run openssl3 with CI somehow (I could then maybe fix possible issues before integration).

No idea sorry. This is not easy because it's a critical system library and right now rimage does not link to it statically. Similar problem: https://thesofproject.github.io/latest/getting_started/build-guide/build-from-scratch.html?highlight=alsa#build-alsa-lib-and-alsa-utils-from-source
> Installing alsa-lib systemwide may break some audio applications. Only perform this if you know what you are doing....

(I never did this myself, instead I just skip the ASRC tests broken by alsalib on Ubuntu 20)

> . One thing we could do is print the version of openssl in the rimage output so that the version is obvious for any breakage.

Yes please.

**Comment by marc-hb** [comment]
> This is not easy because it's a critical system library and right now rimage does not link to it statically. 

This is BTW exactly why most popular programming languages tend to ignore distro packages. Even C/C++ sometimes: https://buckaroo.pm/

https://lwn.net/Articles/867657/ "A longstanding tug-of-war between system package managers and Python's own installation mechanisms (primarily pip, but there are others looks on its way to being resolved—or at least regularized
https://lwn.net/Articles/805818/ (same with Rust)

etc. etc.

tl;dr: software distribution is a hard, unsolved problem
https://medium.com/@sdboyer/so-you-want-to-write-a-package-manager-4ae9c17d9527
